images/airflow/2.9.2/python/mwaa/celery/task_monitor.py

```diff
@@ -136,14 +147,16 @@ def _create_shared_mem_celery_state():
 # will use to signal the toggle of a flag which tells the Celery SQS channel to
 # pause/unpause further consumption of available SQS messages. It is maintained by the
 # worker monitor.
-def _create_shared_mem_work_consumption_block():
+# When MWAA signal handling is enabled, the consumption will be turned off by default and
+# it will be enabled only when the activation signal has been received by the worker.
+def _create_shared_mem_work_consumption_block(mwaa_signal_handling_enabled: bool):
     celery_work_consumption_block_name = (
         f'celery_work_consumption_{os.environ.get("AIRFLOW_ENV_ID", "")}'
     )
     celery_work_consumption_flag_block = shared_memory.SharedMemory(
```

Comment:
Does the SharedMemory need to be unlinked?

Reply:
The code comment in the `close` method explains this. I am pasting it below for reference:
```
This will be called when the worker has been terminated. We close these blocks
        in order to avoid receiving a warning message in the logs. We are not calling
        unlink here intentionally because both the task monitor and the Celery SQS
        channel references these blocks and when the worker is shutting down, we do not
        control which process will be killed first. Calling unlink may result in a
        warning message showing up in the customer side logs causing unnecessary
        confusion.
```